### PR TITLE
Update web API intro doc for 3.0

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/13873
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/14051
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/14279

[Internal Review Page](https://review.docs.microsoft.com/aspnet/core/web-api/index?view=aspnetcore-3.0&branch=pr-en-us-14201)

### CODE REVIEW

File: aspnetcore/web-api/index.md [hunk 1]
```diff
@@ -101,15 +131,25 @@ namespace WebApiSample
...
+
+::: moniker-end
+
+::: moniker range="<= aspnetcore-2.2"
+
+[!code-csharp[](index/samples/2.x/Controllers/ValuesController.cs?name=snippet_ControllerSignature&highlight=1)]
+
+::: moniker-end
 
 Actions are inaccessible via [conventional routes](xref:mvc/controllers/routing#conventional-routing) defined by <xref:Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvc*> or <xref:Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvcWithDefaultRoute*> in `Startup.Configure`.
```
Comment (scottaddie) [inline aspnetcore/web-api/index.md]: @rynowak How does this statement change with the inclusion of endpoint routing in the 3.0 web API project template?

Comment (Rick-Anderson) [reply aspnetcore/web-api/index.md]: @rynowak How does this statement change with the inclusion of endpoint routing in the 3.0 web API project template?
@dougbu  who's the backup for @rynowak  on endpoint routing?

Comment (dougbu) [reply aspnetcore/web-api/index.md]: @rynowak is around. @JamesNK is his backup.

Comment (Rick-Anderson) [reply aspnetcore/web-api/index.md]: @JamesNK How does this statement change with the inclusion of endpoint routing in the 3.0 web API project template?

Comment (JamesNK) [reply aspnetcore/web-api/index.md]: It shouldn't have changed. Endpoint routing is designed to be backwards compatible as much as possible with the older routing API.

Comment (scottaddie) [reply aspnetcore/web-api/index.md]: @JamesNK My question wasn't specific enough. Should the 3.0 version of this text say that actions are inaccessible via conventional routes defined by the following?

```csharp
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllerRoute(
                    name: "default",
                    pattern: "{controller=Home}/{action=Index}/{id?}");
            });
```

Comment (JamesNK) [reply aspnetcore/web-api/index.md]: Yes, the same concept applies. Conventional routes in UseMvc and conventional routes in UseEndpoints are both overridden by attribute routing.

**Pull request merged**